### DISCUSSION TITLE
Preview of asset as media

### DISCUSSION THREAD

**Discussion by lasseaeggen** [sanity-io/sanity#3769] [Sanity Studio Support] [answered]
I'm unable to preview an image as media, when I define a custom type that is of type image.
Since this works in V2, I assume it's supposed to work in V3 too.
I have confirmed that the `asset`, named `media`, is a valid image asset.

```
// Schema that works in V2 - previews the image in the media position
// Does not work in V3
export default defineType({
  type: "image",
  ...,
  preview: {
    select: {
      media: "asset",
    }
}
```

If I try to explicitly set up the preview of `asset` with the MediaPreview/DefaultPreview it still doesn't show an image as the media preview. I might not be using the components correctly, but the result is the same as when using the example code above.
I also tried handling this in the type's `components: {preview: ...}`, but didn't succeed there either, seems like there isn't much difference in that and the good old `prepare()`.

I currently use the following workaround for a Studio under Next:
```
preview: {
  select: {
    media: "asset.url",
  },
  prepare({ media }) {
    return {
      media: () => (
        // eslint-disable-next-line @next/next/no-img-element
        <img src={`${media}?w=35&h=35&fit=crop&dpr=2`} alt="preview" />
      ),
    };
  },
}
```

**Comment by quirm** [comment]
I was trying to do something like this as described in V2 docs:
```
preview: {
    select: {
      title: 'name.en',
      imageUrl 'images[0]->asset.url',
    }
  },
```

After some trial errors, before I discovered types for schema, I ended up with this which is currently working (no need for JSX):
```
preview: {
    select: {
      title: 'name.en',
      images: 'images',
    },
    prepare({ title, images }) {
      return {
        title,
        media: images ? images[0] : null,
      };
    },
  }
```

**Reply by lasseaeggen** [reply]
Thanks for reaching out.
That works if you want to show a preview of the first image in the field of type `array` named `images` in a schema. Then it will use the default image preview component for the media part of the default preview.

What I'm trying to do is preview the image (or `asset` field of Sanity's image object) when the schema is of the type `image`.

**Comment by spacedawwwg** [comment]
I've just come up against this same issue.

Trying to figure out how to get an image preview on a Schema that extends in the in-build image type

**Comment by bjoerge** [comment, marked as answer]
This should be fixed since v3.0.0. Let us know that's not the case.

**Reply by lasseaeggen** [reply]
It's fixed since 3.0.0. Much appreciated.
